client/lib/interpolate-components/Makefile

```diff
@@ -1,7 +1,23 @@
 REPORTER ?= spec
-MOCHA ?= ../../../node_modules/.bin/mocha
+
+ifneq ("$(wildcard node_modules/.bin/)","")
+NODE_BIN ?= ./node_modules/.bin
+else
+NODE_BIN ?= ../../../node_modules/.bin
+endif
+
+BABEL ?= $(NODE_BIN)/babel
+MOCHA ?= $(NODE_BIN)/mocha
 
 test:
 	@NODE_ENV=test NODE_PATH=test:../../../client $(MOCHA) --compilers jsx:babel/register --reporter $(REPORTER)
 
+publish:
+	@npm install
+	@make test
+	$(BABEL) -sd lib/ src/
+	@npm publish
+	rm -rf lib
+	rm -rf node_modules
```

Comment:
You need to NPM install here so that it can test the module in isolation from the rest of the project.  If you let it use the calypso node_modules directory the versions may be different.  Then when the module is published and the consumer installs using the local package.json there could be issues.